### PR TITLE
Update gradle and start using cmake for the native build

### PR DESCRIPTION
**Pull request by szurbrigg**

### CODE REVIEW

File: tests/build.gradle [hunk 1]
```diff
@@ -30,16 +32,18 @@ android {
 
 dependencies {
   compile project(':duktape')
+  androidTestCompile project(path: ':duktape')
+  androidTestCompile project(path: ':duktape')
```
Comment (swankjesse) [inline tests/build.gradle]: drop one of these?

Comment (swankjesse) [inline tests/build.gradle]: Or both?

File: .travis.yml [hunk 2]
```diff
@@ -3,20 +3,20 @@ language: android
 android:
   components:
     - platform-tools
-    - build-tools-23.0.1
```
Comment (JakeWharton) [inline .travis.yml]: 25.0.0 is latest

File: .travis.yml [hunk 3]
```diff
@@ -3,20 +3,20 @@ language: android
...
-    - android-23
+    - build-tools-23.0.3
+    - android-25
     - extra-android-m2repository
-    - sys-img-armeabi-v7a-android-18
+    - sys-img-armeabi-v7a-android-25
```
Comment (JakeWharton) [inline .travis.yml]: The API 25 emulator won't work on Travis, but you're still creating an API 18 emulator below anyway so this should be kept at 18 as well.

File: tests/build.gradle [hunk 4]
```diff
@@ -1,24 +1,26 @@
...
-    classpath 'com.android.tools.build:gradle:2.1.2'
+    classpath 'com.android.tools.build:gradle:2.2.2'
   }
 }
 
-apply plugin: 'com.android.application'
-
 android {
-  compileSdkVersion rootProject.ext.compileSdkVersion
-  buildToolsVersion rootProject.ext.buildToolsVersion
```
Comment (JakeWharton) [inline tests/build.gradle]: Why get rid of the constants?

File: tests/build.gradle [hunk 5]
```diff
@@ -30,16 +32,17 @@ android {
 
 dependencies {
   compile project(':duktape')
+  androidTestCompile project(path: ':duktape')
```
Comment (JakeWharton) [inline tests/build.gradle]: Why is this needed?

File: gradle/wrapper/gradle-wrapper.properties [hunk 6]
```diff
@@ -1,6 +1,6 @@
-#Wed Aug 26 15:08:36 EDT 2015
+#Tue Nov 08 09:37:49 EST 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip
```
Comment (JakeWharton) [inline gradle/wrapper/gradle-wrapper.properties]: 3.1 is latest

File: duktape/build.gradle [hunk 7]
```diff
@@ -1,124 +1,56 @@
...
-model {
-  android {
-    compileSdkVersion = rootProject.ext.compileSdkVersion
-    buildToolsVersion = rootProject.ext.buildToolsVersion
-
-    defaultConfig {
-      minSdkVersion.apiLevel = 4
+android {
+  compileSdkVersion 25
+  buildToolsVersion '23.0.3'
```
Comment (JakeWharton) [inline duktape/build.gradle]: constants

File: duktape/build.gradle [hunk 8]
```diff
@@ -1,124 +1,56 @@
...
+  defaultConfig {
+    minSdkVersion 15
```
Comment (JakeWharton) [inline duktape/build.gradle]: This used to be 4, is there any reason to bump it?

File: .travis.yml (hunk 2 above)
Comment (szurbrigg) [inline .travis.yml]: changed

File: .travis.yml (hunk 3 above)
Comment (szurbrigg) [inline .travis.yml]: reverted

File: tests/build.gradle (hunk 4 above)
Comment (szurbrigg) [inline tests/build.gradle]: restored

File: tests/build.gradle (hunk 5 above)
Comment (szurbrigg) [inline tests/build.gradle]: This makes the JUnit tests work.

File: duktape/build.gradle (hunk 8 above)
Comment (szurbrigg) [inline duktape/build.gradle]: I spent a lot of time stabbing at a bunch of different things yesterday trying to get a build...  I'll change this back.

File: gradle/wrapper/gradle-wrapper.properties (hunk 6 above)
Comment (szurbrigg) [inline gradle/wrapper/gradle-wrapper.properties]: changed

File: duktape/build.gradle (hunk 8 above)
Comment (szurbrigg) [inline duktape/build.gradle]: Turns out this was needed for the native part of the build?  I can go as low as 12 and have it still work.

File: .travis.yml [hunk 9]
```diff
@@ -2,21 +2,35 @@ language: android
 
 android:
   components:
+    - tools
     - platform-tools
-    - build-tools-23.0.1
-    - android-23
+    - tools
```
Comment (baoti) [inline .travis.yml]: Seem to duplicate

Comment (szurbrigg) [inline .travis.yml]: Yes.  This tricks travis into accepting the license for the SDK.  See https://github.com/travis-ci/docs-travis-ci-com/issues/779.
